lib/doorkeeper/config.rb

```diff
@@ -7,54 +7,38 @@ def initialize
 
   def self.configure(&block)
     @config = Config::Builder.new(&block).build
-    enable_orm
-    check_for_missing_columns
+    setup_orm_adapter
+    setup_orm_models
     setup_application_owner if @config.enable_application_owner?
+    check_requirements
   end
 
   def self.configuration
     @config || (fail MissingConfiguration.new)
   end
 
-  def self.check_for_missing_columns
-    if Doorkeeper.configuration.orm == :active_record &&
-        ActiveRecord::Base.connected? &&
-        ActiveRecord::Base.connection.table_exists?(
-          Doorkeeper::Application.table_name
-        ) &&
-        !Doorkeeper::Application.new.attributes.include?("scopes")
-
-      puts <<-MSG.squish
-[doorkeeper] Missing column: `oauth_applications.scopes`.
-If you are using ActiveRecord run `rails generate doorkeeper:application_scopes
-&& rake db:migrate` to add it.
-      MSG
-    end
+  def self.check_requirements
+    @orm_adapter.check_requirements!(@config)
   end
 
-  def self.enable_orm
-    class_name = "doorkeeper/orm/#{configuration.orm}".classify
-    class_name.constantize.initialize_models!
-  rescue NameError => e
-    if e.instance_of?(NameError)
-      fail e, "ORM adapter not found (#{configuration.orm})", <<-error_msg
+  def self.setup_orm_adapter
+    @orm_adapter = "doorkeeper/orm/#{configuration.orm}".classify.constantize
+  rescue => e
```

Comment:
Does this catch _any_ exception? Which are the ones we care about? Can the code raise other type of exceptions?